lib/pure/strutils.nim

```diff
@@ -2319,6 +2319,46 @@ proc removeSuffix*(s: var string, suffix: string) {.
     newLen -= len(suffix)
     s.setLen(newLen)
 
+proc removePrefix*(s: var string, chars: set[char] = Newlines) {.
+  rtl, extern: "nsuRemovePrefixCharSet".} =
+  ## Removes all characters from `chars` from the start of the string `s`
+  ## (in-place).
+  ## .. code-block:: nim
+  ##   var userInput = "\r\n*~Hello World!"
+  ##   userInput.removePrefix
+  ##   userInput == "*~Hello World!"
```

Comment:
Make examples like this compilable, so use ``doAssert`` here. It makes things clearer and ensures we can test these examples in the future.